pagespeed/kernel/http/response_headers.cc

```diff
@@ -321,23 +321,28 @@ bool ResponseHeaders::CombineContentTypes(const StringPiece& orig,
 }
 
 bool ResponseHeaders::MergeContentType(const StringPiece& content_type) {
-  bool ret = false;
+  bool ret = std::find_if(
+      content_type.begin(), content_type.end(), iscntrl) == content_type.end();
   ConstStringStarVector old_values;
-  Lookup(HttpAttributes::kContentType, &old_values);
-  // If there aren't any content-type headers, we can just add this one.
-  // If there is exactly one content-type header, then try to merge it
-  // with what we were passed.
-  // If there is already more than one content-type header, it's
-  // unclear what exactly should happen, so don't change anything.
-  if (old_values.size() < 1) {
-    ret = CombineContentTypes("", content_type);
-  } else if (old_values.size() == 1) {
-    StringPiece old_val(*old_values[0]);
-    ret = CombineContentTypes(old_val, content_type);
-  }
+
   if (ret) {
-    cache_fields_dirty_ = true;
+    Lookup(HttpAttributes::kContentType, &old_values);
```

Comment:
@crowell  This is exactly like it was before this change. Double checking, it looks to me like this should work ok when `Lookup` would fail (`old_values` should be unchanged). Do you have additional handling in mind for the case where `Lookup` returns false?